program/llvm-toolchain-ppl.md

```diff
@@ -0,0 +1,252 @@
+# OpenHW Preliminary Project Proposal: LLVM tool chain for CORE-V
+
+## Summary of project
+
+This proposal is for a baseline development tool chain for CORE-V based on up-to-date compiler and simulation technology, comprising:
+
+- Clang/LLVM tool chain for C and C++:
+  - compiler
+  - assembler
+  - linker
+  - emulation library (`compiler-rt`)
+
+The CORE-V LLVM tool chain may rely initially on the GNU CORE-V binutils developed as part of the GNU CORE-V tool chain:
+  - assembler (`gas`)
+  - linker (GNU `ld`)
+  - low level utilities for manipulating object files.
+
+The tool chain will interface to the following other components of the CORE-V development ecosystem:
+- GNU debugger (GDB)
+- standard C libraries (`newlib`, GNU `_GlibC_`)
+- GNU C++ standard library (`libstdc++v3`)
+
+NOTE: Provision of non-(L)GPL C/C++ libraries is beyond the scope of this project.
+
+Because of the scale of the project and the current maturity level of LLVM for RISC-V in general, the LLVM-related work is decomposed into three phases:
+
+1. Support for bare metal use of C with CVA6 (64-/32-bit) together with the _Newlib_ C library;
+2. Support for Linux application use of C and C++ with CVA6 (32-/64-bit) together with the _GlibC_ C library; and
+3. Support for other RTOSes.
+
+The proposal addresses just the first phase. Other phases will be the subject of separate proposals. Phase 1 will provide the following:
+
+- suitability of Clang/LLVM for HW verification of CVA6:
+  - seamless support of multiple datapath widths (multi-XLEN support);
```

Comment:
This already exists upstream. `riscv32-unknown-elf-clang` can compile RV64, it just defaults to RV32. Similarly riscv64-`unknown-elf-clang` can compile RV32, it just defaults to RV64. Did you mean more than this?